### PR TITLE
Add linked schema resolution

### PR DESCRIPTION
**Pull request by skiddykong**

### CODE REVIEW

File: aggregate-snapshot/aggregate-snapshot-repository/pom.xml [hunk 1]
```diff
@@ -44,5 +44,10 @@
             <version>${project.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>test-utils-core</artifactId>
+            <version>2.3.0-SNAPSHOT</version>
```
Comment (joaoveigacap) [inline aggregate-snapshot/aggregate-snapshot-repository/pom.xml]: add   <scope>test</scope> ?

File: aggregate-snapshot/aggregate-snapshot-repository/src/test/java/uk/gov/justice/services/eventsourcing/jdbc/snapshot/SnapshotRepositoryJdbcIT.java [hunk 2]
```diff
@@ -87,7 +88,8 @@ public void shouldRetrieveLatestSnapshotWithCorrectType() {
         jdbcRepository.storeSnapshot(aggregateSnapshot2);
         jdbcRepository.storeSnapshot(aggregateSnapshot3);
 
-        final Optional<AggregateSnapshot<RecordingAggregate>> snapshot = jdbcRepository.getLatestSnapshot(STREAM_ID, TYPE);
+        Poller poller = new Poller();
+        final Optional<AggregateSnapshot<RecordingAggregate>> snapshot  = poller.pollUntilFound(() -> jdbcRepository.getLatestSnapshot(STREAM_ID, TYPE));
```
Comment (purple52) [inline aggregate-snapshot/aggregate-snapshot-repository/src/test/java/uk/gov/justice/services/eventsourcing/jdbc/snapshot/SnapshotRepositoryJdbcIT.java]: Is this change related to this pull request?

File: core/pom.xml [hunk 3]
```diff
@@ -52,8 +52,15 @@
             <artifactId>commons-lang3</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.everit.json</groupId>
+            <groupId>com.github.everit-org.json-schema</groupId>
             <artifactId>org.everit.json.schema</artifactId>
+            <version>${org.everit.json.schema.version}</version>
```
Comment (purple52) [inline core/pom.xml]: This needs to move into the common BOM

File: core/src/main/java/uk/gov/justice/services/core/json/DefaultJsonSchemaValidatorFactory.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package uk.gov.justice.services.core.json;
+
+import org.slf4j.LoggerFactory;
+
+public class DefaultJsonSchemaValidatorFactory {
```
Comment (purple52) [inline core/src/main/java/uk/gov/justice/services/core/json/DefaultJsonSchemaValidatorFactory.java]: Shouldn't this class be in Test not Main?

File: event-sourcing/event-source-api-integration-test/conf/conf.d/cxf-rs.properties [hunk 5]
```diff
@@ -0,0 +1,3 @@
+server      = org.apache.openejb.server.cxf.rs.CxfRSService
```
Comment (purple52) [inline event-sourcing/event-source-api-integration-test/conf/conf.d/cxf-rs.properties]: Not sue these files in "conf" should be part of the commit... shouldn't they be excluded via .gitignore?

File: example-context/example-service/example-it/src/test/resources/wildfly-config/standalone-single.xml [hunk 6]
```diff
@@ -442,6 +442,7 @@
                 <simple name="java:global/eventlog.insertion.strategy" value="uk.gov.justice.services.eventsourcing.repository.jdbc.AnsiSQLEventLogInsertionStrategy" type="java.lang.String"/>
                 <simple name="java:global/envelope.validation.exception.handler" value="uk.gov.justice.services.core.envelope.RethrowingValidationExceptionHandler" type="java.lang.String"/>
                 <simple name="java:global/fileservice.metadata.sql.provider.strategy" value="uk.gov.justice.services.fileservice.repository.AnsiMetadataSqlProvider" type="java.lang.String"/>
+                <simple name="java:global/file.system.url.resolver.strategy" value="uk.gov.justice.services.core.json.VfsUrlResolverStrategy" type="java.lang.String"/>
```
Comment (purple52) [inline example-context/example-service/example-it/src/test/resources/wildfly-config/standalone-single.xml]: Is this needed if we're using @Alternative?

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/it/JmsEndpointGenerationIT.java [hunk 7]
```diff
@@ -83,8 +114,10 @@
             TestServiceContextNameProvider.class,
             DefaultJmsMessageLoggerHelper.class,
             DefaultTraceLogger.class,
-            DefaultJsonValidationLoggerHelper.class
+            DefaultJsonValidationLoggerHelper.class,
 
+            DefaultFileSystemUrlResolverStrategy.class,
+            GlobalValueProducer.class,
```
Comment (purple52) [inline generators/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/it/JmsEndpointGenerationIT.java]: Is this needed if we're using @Alternative?

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultCustomUserIdResourceIT.java [hunk 8]
```diff
@@ -131,7 +134,10 @@ public Properties properties() {
             ResponseStrategyCache.class,
             ValidParameterCollectionBuilderFactory.class,
             DefaultTraceLogger.class,
-            DefaultHttpTraceLoggerHelper.class
+            DefaultHttpTraceLoggerHelper.class,
+
+            DefaultFileSystemUrlResolverStrategy.class,
+            GlobalValueProducer.class
```
Comment (purple52) [inline generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultCustomUserIdResourceIT.java]: Still needed?

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultUsersUserIdResourceIT.java [hunk 9]
```diff
@@ -150,7 +153,10 @@ public Properties properties() {
             ResponseStrategyCache.class,
             ValidParameterCollectionBuilderFactory.class,
             DefaultTraceLogger.class,
-            DefaultHttpTraceLoggerHelper.class
+            DefaultHttpTraceLoggerHelper.class,
+
+            DefaultFileSystemUrlResolverStrategy.class,
+            GlobalValueProducer.class
```
Comment (purple52) [inline generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultUsersUserIdResourceIT.java]: Still needed?

File: pom.xml [hunk 10]
```diff
@@ -53,7 +54,8 @@
         <utilities.version>1.7.0</utilities.version>
         <test-utils.version>1.5.0</test-utils.version>
         <jboss-ejb3-ext-api.version>2.2.0.Final</jboss-ejb3-ext-api.version>
-
+        <jboss-vfs.version>3.2.12.Final</jboss-vfs.version>
+        <org.everit.json.schema.version>1.6.0</org.everit.json.schema.version>
```
Comment (purple52) [inline pom.xml]: These should go into the common BOM

File: aggregate-snapshot/aggregate-snapshot-repository/src/test/java/uk/gov/justice/services/eventsourcing/jdbc/snapshot/SnapshotRepositoryJdbcIT.java (hunk 2 above)
Comment (skiddykong) [reply aggregate-snapshot/aggregate-snapshot-repository/src/test/java/uk/gov/justice/services/eventsourcing/jdbc/snapshot/SnapshotRepositoryJdbcIT.java]: flaky test was failing my local build intermittently

File: core/pom.xml (hunk 3 above)
Comment (skiddykong) [reply core/pom.xml]: ok

File: event-sourcing/event-source-api-integration-test/conf/conf.d/cxf-rs.properties (hunk 5 above)
Comment (skiddykong) [reply event-sourcing/event-source-api-integration-test/conf/conf.d/cxf-rs.properties]: I'll add them

File: example-context/example-service/example-it/src/test/resources/wildfly-config/standalone-single.xml (hunk 6 above)
Comment (skiddykong) [reply example-context/example-service/example-it/src/test/resources/wildfly-config/standalone-single.xml]: No, I will remove

File: core/src/main/java/uk/gov/justice/services/core/json/DefaultJsonSchemaValidatorFactory.java (hunk 4 above)
Comment (skiddykong) [reply core/src/main/java/uk/gov/justice/services/core/json/DefaultJsonSchemaValidatorFactory.java]: Moved to test-core

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/it/JmsEndpointGenerationIT.java (hunk 7 above)
Comment (skiddykong) [reply generators/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/it/JmsEndpointGenerationIT.java]: removed

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultCustomUserIdResourceIT.java (hunk 8 above)
Comment (skiddykong) [reply generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultCustomUserIdResourceIT.java]: removed

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultUsersUserIdResourceIT.java (hunk 9 above)
Comment (skiddykong) [reply generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultUsersUserIdResourceIT.java]: removed

File: pom.xml (hunk 10 above)
Comment (skiddykong) [reply pom.xml]: moved

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/it/JmsEndpointGenerationIT.java [hunk 11]
```diff
@@ -8,21 +8,52 @@
...
+import uk.gov.justice.services.core.dispatcher.SystemUserUtil;
+import uk.gov.justice.services.core.envelope.EnvelopeValidationExceptionHandlerProducer;
+import uk.gov.justice.services.core.enveloper.Enveloper;
+import uk.gov.justice.services.core.extension.BeanInstantiater;
+import uk.gov.justice.services.core.extension.ServiceComponentScanner;
+import uk.gov.justice.services.core.interceptor.InterceptorCache;
+import uk.gov.justice.services.core.interceptor.InterceptorChainObserver;
+import uk.gov.justice.services.core.interceptor.InterceptorChainProcessor;
+import uk.gov.justice.services.core.interceptor.InterceptorChainProcessorProducer;
+import uk.gov.justice.services.core.json.DefaultFileSystemUrlResolverStrategy;
```
Comment (purple52) [inline generators/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/it/JmsEndpointGenerationIT.java]: What changed to cause all these imports to be added?

File: generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultUsersUserIdResourceIT.java [hunk 12]
```diff
@@ -31,11 +31,14 @@
 import uk.gov.justice.services.adapter.rest.processor.response.OkStatusEnvelopeEntityResponseStrategy;
 import uk.gov.justice.services.adapter.rest.processor.response.OkStatusEnvelopePayloadEntityResponseStrategy;
 import uk.gov.justice.services.adapter.rest.processor.response.ResponseStrategyHelper;
+import uk.gov.justice.services.common.configuration.GlobalValueProducer;
```
Comment (purple52) [inline generators/rest-adapter-generator/src/test/java/uk/gov/justice/api/resource/DefaultUsersUserIdResourceIT.java]: Unused import?

File: event-sourcing/event-source-api-integration-test/conf/conf.d/cxf-rs.properties (hunk 5 above)
Comment (purple52) [reply event-sourcing/event-source-api-integration-test/conf/conf.d/cxf-rs.properties]: The files themselves are still in the commit, though, and need removing

File: generators/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/it/JmsEndpointGenerationIT.java (hunk 11 above)
Comment (skiddykong) [reply generators/messaging-adapter-generator/src/test/java/uk/gov/justice/raml/jms/it/JmsEndpointGenerationIT.java]: When I was trying to figure out what was missing to use the global jndi variables - will clean
